blocks/library/list/index.js

```diff
@@ -72,6 +112,17 @@ registerBlock( 'core/list', {
 		return (
 			<Editable
 				tagName={ nodeName.toLowerCase() }
+				onConfig={ ( settings ) => ( {
```

Comment:
I like that this is immutable, but it does seem a bit odd to have an `on`- prefixed callback expected to return a value. I think we could choose between keeping this as `onConfig` and mutating `settings`, or changing the name to something like `getSettings`.

Or, and forgive me again for my extreme pondering, explore functions as children as an interesting extensibility option for overriding the default TinyMCE rendering to extend settings. Something like:

```jsx
<Editable
	{ /* ... */ }
>
	( tinyMCEProps ) => (
		<TinyMCE 
			{ ...tinyMCEProps }
			settings={
				...tinyMCEProps.settings,
				plugins: ( settings.plugins || [] ).concat( 'lists' ),
				lists_indent_on_tab: false,
			}
		/>
	)
</Editable>
```